x/ecocredit/server/msg_server.go

```diff
@@ -295,30 +287,25 @@ func (s serverImpl) Retire(goCtx context.Context, req *ecocredit.MsgRetireReques
 	return &ecocredit.MsgRetireResponse{}, nil
 }
 
-func (s serverImpl) retire(ctx sdk.Context, store sdk.KVStore, recipient string, batchDenom batchDenomT, retired *apd.Decimal) error {
-	err := getAddAndSetDecimal(store, RetiredBalanceKey(recipient, batchDenom), retired)
+func (s serverImpl) SetPrecision(goCtx context.Context, req *ecocredit.MsgSetPrecisionRequest) (*ecocredit.MsgSetPrecisionResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	classID, err := ecocredit.BatchID2ClassID(req.BatchDenom)
 	if err != nil {
-		return err
+		return nil, err
+	}
+	if err = s.assertIssuer(ctx, classID, req.Issuer); err != nil {
+		return nil, err
```

Comment:
We just want to check that the issuer is the issuer of the batch. It should not be possible for any issuer to change the precision. Just the one who issued the batch. Based on this I don't see a need for `BatchID2ClassID` and `s.assertIssuer`. The only time this sort of check is needed is in `CreateBatch` and it doesn't need to be factored out into a separate function.

Reply:
I removed `BatchID2ClassID`, but I keep the `assertIssuer` - it's easier to read the code if we can break down some functions.

Reply:
And I've updated the code as per your the spec here.